File: ProyectoCsharp/Modelo/AlmacenVirtual.cs
using System;
using System.Collections.Generic;

namespace Modelo
{
    public class AlmacenVirtual : Almacen
    {
        private List<Area> areas;
        //Constructores
        public AlmacenVirtual()
        {
            areas = new List<Area>();
        }

        public AlmacenVirtual(long id, string nom, string descripcion) : base(id, nom, descripcion)
        {
            areas = new List<Area>();
        }

        //DEstructor
        ~AlmacenVirtual()
        {
            areas = null;
        }

        public void AgregarArea(Area areas)
        {
            this.areas.Add(areas);
        }

        public List<Area> GetArea()
        {
            return this.areas;
        }
    }

}

File: TrabajoJavaLp2/src/modelo/AlmacenVirtual.java
package modelo;
import java.util.ArrayList;
public class AlmacenVirtual extends Almacen{
	private ArrayList<Area> areas;
	public AlmacenVirtual(){
		
	}

	public AlmacenVirtual(long id, String nom){
		super(id,nom);
	}
	public void setArea(Area areas){
		this.areas.add(areas);
	}

	public ArrayList<Area> getArea(){
		return areas;
	}
}